File: library/src/main/java/io/ashdavies/rx/rxtasks/CompletableEmitterFailureListener.java
package io.ashdavies.rx.rxtasks;

import android.support.annotation.NonNull;
import com.google.android.gms.tasks.OnFailureListener;
import io.reactivex.CompletableEmitter;

class CompletableEmitterFailureListener implements OnFailureListener {

  private final CompletableEmitter emitter;

  CompletableEmitterFailureListener(CompletableEmitter emitter) {
    this.emitter = emitter;
  }

  @Override
  public void onFailure(@NonNull Exception exception) {
    emitter.onError(exception);
  }
}

File: library/src/main/kotlin/io/ashdavies/rx/rxtasks/CompletableEmitterFailureListener.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnFailureListener
import io.reactivex.CompletableEmitter

internal class CompletableEmitterFailureListener(private val emitter: CompletableEmitter) : OnFailureListener {

  override fun onFailure(exception: Exception) = emitter.onError(exception)
}
